client/object_put.go

```diff
@@ -325,3 +334,173 @@ func (c *Client) ObjectPutInit(ctx context.Context, hdr object.Object, signer us
 
 	return &w, nil
 }
+
+// ReplicateObject copies binary-encoded NeoFS object from the given
+// [io.ReadSeeker] to remote server for local storage. The signer must
+// authenticate a storage node that matches the object's storage policy. Since
+// this property can change over NeoFS system time, compliance with the policy
+// is checked back to foreseeable moment in the past. The server must be a
+// storage node compliant with the current object's storage policy.
+//
+// ReplicateObject is intended for maintaining data storage by NeoFS system
+// nodes only, not for regular use.
+//
+// Object must be encoded in compliance with Protocol Buffers v3 format in
+// ascending order of fields.
+//
+// Source [io.ReadSeeker] must point to the start. Note that ReplicateObject
+// does not reset src to start after the call. If it is needed, do not forget to
+// Seek.
+//
+// Return errors:
+//   - [apistatus.ErrServerInternal]: internal server error described in the text message;
+//   - [apistatus.ErrObjectAccessDenied]: the signer does not authenticate any
+//     NeoFS storage node matching storage policy of the container referenced by the
+//     replicated object;
+//   - [apistatus.ErrContainerNotFound]: the container to which the replicated
+//     object is associated was not found.
+func (c *Client) ReplicateObject(ctx context.Context, src io.ReadSeeker, signer neofscrypto.Signer) error {
```

Comment:
-1 for building objects in SDK. that is what api-go is created for to me

Reply:
> building objects

u mean messages (cuz NeoFS objects are not "built" within this scope)? api-go lib doesnt provide functionality for this, so i implemented it in-place

Reply:
> u mean messages (cuz NeoFS objects are not "built" within this scope)?

yes. the same way SDK does not have any `Marshal()` func implementation. SDK's `go.mod` does not have any explicit proto/grpc lib. why now? because of some new operation?

Reply:
> SDK's go.mod does not have any explicit proto/grpc lib

this does not depend on this PR, anyway these packages are required ( `// indirect` remark doesn't mean that much)

> extend api-go then?

this should be motivated (and can be done at any time btw). I agree with @roman-khimov that right now `no ones uses except for SDK and no one will`

Reply:
> this does not depend on this PR

this _does_ depend on this PR cause it adds them

> // indirect remark doesn't mean that much

it means a lot. it explicitly says that this repo does not use it and there is no code about it. SDK uses API, API uses proto/grpc. now SDK uses grpc

> this should be motivated

proto things should be done in a proto repo? not enough?

> I agree with @roman-khimov that right now no ones uses except for SDK and no one will

how does it relate to my comment? we can have `ObjectReplicateInit` in SDK and `ReplicateObject` in API the same way how it has been done for _every_ method before. if we need some additional helper func that can build messages based on raw bytes, then we can add them in the API, no problem. how has a new RPC become a reason for a cross-repo movement?